tests/test_dna/test_dna_feature_extraction.py

```diff
@@ -216,4 +216,44 @@ def test_extract_dna_features():
         # Removing the features calculated from the input csvs to facilitate future tests
         df = pd.read_csv(extracted_features_filepath)
         df = df.iloc[:, :7]
-        df.to_csv(extracted_features_filepath, index=False)
\ No newline at end of file
+        df.to_csv(extracted_features_filepath, index=False)
+
+
+def test_extract_dna_features_unexpected_input():
+
+    input_folder_path = os.path.join(
+        os.path.dirname(__file__), "feature_extraction_bad_input"
+    )
+
+    # Extract features from input csv files in input folder path
+    dna_feature_extraction.extract_dna_features(input_folder_path)
+
+    try:
+        # Extract the filenames of the txt files where the gene id lists are stored
+        extracted_features_filepath = (
+                f"{input_folder_path}/ensembl_data_chlamydomonas_reinhardtii.csv"
+            )
+    except FileNotFoundError:
+        print(
+            f"Test input file ensembl_data_chlamydomonas_reinhardtii.csv not found."
+        )
+
+    with open(extracted_features_filepath) as file:
+        reader = csv.DictReader(file)
+        features_extracted = [gene for gene in reader]
+
+    correct_headers = [
+        "ensembl_gene_id",
+        "transcript_id",
+        "promoter",
+        "utr5",
+        "cds",
+        "utr3",
+        "terminator"
+    ]
+
+    for gene in features_extracted:
```

Comment:
Here you're looping through the whole file line by line. I think just looping through the header should be enough, no?